File: src/app/models/TwinChangeEvent.js


// Define collection and schema for TwinChangeEvent
export interface TwinChangeEvent {
    eventId:
	type : string
    occurredAt:
	type : Date
    Twin:
	type : Schema.Types.ObjectId
    ChangeType:
 	type : String
#
    collection: 'twinChangeEvents'
}

File: iotonaspdotnet/Domain/TwinChangeEvent.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class TwinChangeEvent
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long TwinchangeeventId { get; set; }
 public virtual string EventId { get; set; }
 public virtual DateTime OccurredAt { get; set; }
public virtual DigitalTwin Twin { get; set; }
 public virtual TwinChangeType ChangeType { get; set; }

    public static TwinChangeEvent FromRequest(TwinChangeEventRequest request) {
        return new TwinChangeEvent {
            Id = request.Id,
            EventId = request.EventId,
            OccurredAt = request.OccurredAt,
            ChangeType = request.ChangeType,
        };
    }
}
